### PR TITLE
Refactored config parsing and flag parsing

### PR DESCRIPTION
**Pull request by ofey404**
Previous flag parsing is a mess! (I'm sorry for that...)

I just refactored the `cmd/` package, made config and flag parsing clearer.

Also add some comments.

**Tested with two configs on a tiup playground**

`should-fail.yaml`:
```yaml
start-after: 0s
interval: 10s
rules:
    - tag: uptime
      promql: rate(process_start_time_seconds{tidb_cluster="", job="tikv"}[1m]) == 0
```

`should-pass.yaml`
```yaml
start-after: 0s
interval: 10s
rules:
    - tag: uptime
      promql: rate(process_start_time_seconds{tidb_cluster="", job="tikv"}[1m]) != 0
```

### CODE REVIEW

File: cmd/root.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+	}
+}
+
+func init() {
+	cobra.OnInitialize(InitConfig)
+	rootCmd.SetOut(os.Stdout)
+
+	rootCmd.PersistentFlags().StringVarP(&Flag.prometheusAPIURL, "address", "u", "http://127.0.0.1:9090", "Host and port of prometheus")
+	rootCmd.PersistentFlags().StringVarP(&Flag.configFilePath, "config", "c", "./config.yaml", "Set config file path, overrided by --config-base64")
+	rootCmd.PersistentFlags().StringVar(&Flag.configBase64, "config-base64", "", "Pass config file as base64 string, override --config")
```
Comment (cosven) [inline cmd/root.go:68]: I found that we can pass text block with following syntax

```
cmd -x x -y="$(< complicated.yaml)"
```

File: cmd/config.go [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+		if err != nil {
+			log.Fatal(err.Error())
+		}
+		Conf = LoadConfigFromBytes(configString)
+		log.Info("Load config from base64 string")
+	}
+	Conf.startTime = time.Now()
+
+	if len(Conf.Rules) == 0 {
+		log.Fatal("Number of rules == 0")
```
Comment (cosven) [inline cmd/config.go]:
```suggestion
		log.Fatal("Number of rules must not be 0")
```

File: pkg/metrics/checker.go [hunk 3]
```diff
@@ -34,7 +34,7 @@ func NewChecker(promAddress string, rules []Rule, interval time.Duration) (*Chec
 // 		 We use single-threaded code here, should improve it.
 // 		 Ref: https://github.com/prometheus/prometheus/blob/19c190b406c992278aaade63be92ecc7bb6a4921/rules/manager.go#L910
 
-// CheckGiven checks whether a given promQL returns true or not.
+// CheckGiven checks whether a given promQL Satisified.
```
Comment (cosven) [inline pkg/metrics/checker.go]:
```suggestion
// CheckGiven checks whether a given promQL is satisfied.
```

File: cmd/root.go (hunk 1 above)
Comment (ofey404) [reply cmd/root.go:68]: Yes. This style is so...Unix 😂

**Pull request merged**